resources/leiningen/new/luminus/oauth/src/oauth.clj

```diff
@@ -0,0 +1,44 @@
+(ns <<project-ns>>.oauth
+  (:require [oauth.client :as oauth]
+            [config.core :refer [env]]
+            [clojure.tools.logging :as log]))
+
+;;Twitter used as an example, replace these URIs with the OAuth provider of your choice
+
+(def request-token-uri
+  "https://api.twitter.com/oauth/request_token")
+
+(def access-token-uri
+  "https://api.twitter.com/oauth/access_token")
+
+(def authorize-uri
+  "https://api.twitter.com/oauth/authenticate")
+
+(def consumer
+  (oauth/make-consumer (env :oauth-consumer-key)
+                       (env :oauth-consumer-secret)
+                       request-token-uri
+                       access-token-uri
+                       authorize-uri
+                       :hmac-sha1))
+
+(defn oauth-callback-uri
+  "Generates the oauth request callback URI"
+  [{:keys [headers]}]
+  (str (or (headers "x-forwarded-proto") "http") "://" (headers "host") "/oauth/twitter-callback"))
```

Comment:
probably best to stick with `https` in the template as it's a better practice :)